### PR TITLE
Restore Box.framework as a dependency

### LINKED ISSUE #36: Restore Box.framework as a dependency
`Result` as a `struct` is [awkward to use](https://github.com/ReactiveCocoa/ReactiveCocoa/pull/1933#discussion_r28931968) and having a public `Box` that isn’t a separate dependency will only lead to ambiguities.

So per discussion in that PR, we’ll restore `Box` to Box.framework and `Result` will be an `enum` once more.

### PR DESCRIPTION
**Pull request by mdiep**
Fixes #36

@robrix Let me know how this looks. I’m happy to make changes! :smile:

### CODE REVIEW

File: Result.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -472,6 +477,11 @@
 				DYLIB_COMPATIBILITY_VERSION = 1;
 				DYLIB_CURRENT_VERSION = 1;
 				DYLIB_INSTALL_NAME_BASE = "@rpath";
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(inherited)",
+					"$(PROJECT_DIR)/Carthage/Checkouts/Box/build/Debug",
+					"$(PROJECT_DIR)/Carthage/Checkouts/Box/build/Debug-iphoneos",
+				);
```
Comment (robrix) [inline Result.xcodeproj/project.pbxproj]: I suspect these two are the Mac framework & tests and should just be Debug, not Debug-iphoneos.

**Comment by robrix** [comment]
You are a kind soul, @mdiep :bow:

File: Result.xcodeproj/project.pbxproj (hunk 1 above)
Comment (mdiep) [inline Result.xcodeproj/project.pbxproj]: I don't think this is needed at all.

I forgot to check if Xcode added any silly search paths. :disappointed:

**Comment by mdiep** [comment]
:tada:

**Comment by robrix** [comment]
:bow: thank you!

**Comment by mdiep** [comment]
Happy to help. :smile:

**Pull request merged**